packages/terra-dynamic-grid/src/DynamicGrid.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { StyleSheet, css } from 'aphrodite';
+import PropTypes from 'prop-types';
+import generateStyles from './styles';
+
+const DynamicGrid = (props) => {
+  const stylesheets = props.layouts
+    .map(generateStyles)
+    .map(StyleSheet.create);
+
+  const layoutStyles = name => stylesheets.map(x => x[name]);
+  return (<div className={css(layoutStyles('grid'))}>
+    {React.Children.map(props.children,
+     region => React.cloneElement(region, {
+       styles: layoutStyles(region.props.name),
```

Comment:
Could you add a code comment about this, took me a while to grok what all this was doing here.